Add feature to save game data.

apetrai opened the pull request:
- Added a new type called GameState which saves the data such as the player position and the zoom.
 - Added member function saveGameState and loadGameState which save and load from the file savegame.dat respectively.
 - Added new parameter in the GameLogic::update() function to take in a reference of GameState, to save and load the values mentioned.
 - Added an exit button.

apetrai committed:
Add feature to process data files

 - Added a new type called GameState which saves the data such as the
   player position and the zoom.
 - Added member function saveGameState and loadGameState which save and
   load from the file savegame.dat respectively.
 - Added new parameter in the GameLogic::update() function to take in a
   reference of GameState, to save and load the values mentioned.
 - Added an exit button.

lemoncho57 commented:
In the save and load function I think it'll be better to do an assert instead of if statement when checking if the file is not opened

apetrai commented:
@lemoncho57 
That's a good idea, I will implement that. If the file is not saved, it will prevent the user from playing the game reducing the potential loss of game data.

lemoncho57 commented:
@apetrai the asserts are already implemented  just use the permaAssert or permaAssertComment macros

apetrai committed:
Add permaAsserts to load and save function.

Also added a feature preventing a user from playing the game if the
filesystem is not working properly - meaning that if the files are not
saving properly the program will be aborted before the user can play.

apetrai committed:
Fix typo in code
